### PR TITLE
Add `load_study` function.

### PR DESCRIPTION
**Pull request by sile**
Currently, `optuna.study.Study()` constructor can be used to load existing studies. 
However, since this behavior is a bit confusing, this PR adds a more intuitive function `optuna.study.load_study()` for this purpose.

# Generated Documentation

image

image

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -511,14 +503,60 @@ def create_study(
...
+    if _direction == structs.StudyDirection.MAXIMIZE:
+        raise ValueError('Optimization direction of study {} is set to `MAXIMIZE`. '
+                         'Currently, Optuna supports `MINIMIZE` only.'.format(study_name))
+
+    study.storage.set_study_direction(study_id, _direction)
+
+    return study
+
+
+def load_study(
```
Comment (g-votte) [inline optuna/study.py]: Would you mind adding a unit test like this? https://github.com/pfnet/optuna/blob/master/tests/test_study.py#L470

File: optuna/study.py [hunk 2]
```diff
@@ -33,6 +33,11 @@ class Study(object):
     This object provides interfaces to run a new :class:`~optuna.trial.Trial`, access trials'
     history, set/get user-defined attributes of the study itself.
 
+    For creating and loading studies, please refer to the documentation of
+    :func:`~optuna.study.create_study` and :func:`~optuna.study.load_study` respectively.
+
+    Note that the direct use of this constructor is not recommended.
```
Comment (g-votte) [inline optuna/study.py]: In my opinion, this sentence had better be presented in an upper line. E.g., just after `A study corresponds to an optimization task, i.e., a set of trials.`

Comment (sile) [reply optuna/study.py]: Sounds good. I will move it.

File: optuna/study.py (hunk 1 above)
Comment (sile) [reply optuna/study.py]: Thank you for your information. I will add a unit test.

File: optuna/study.py (hunk 2 above)
Comment (sile) [reply optuna/study.py]: Moved: https://github.com/pfnet/optuna/pull/330/commits/b0cfea8d5caed1bb25ab54717260ce73f118c47c

File: optuna/study.py (hunk 1 above)
Comment (sile) [reply optuna/study.py]: Added: https://github.com/pfnet/optuna/pull/330/commits/a9a08014aa79d4e05f7b25cf9db040b7cbc3ea69

**Comment by toshihikoyanase** [comment]
Could you add the following instruction to `optuna.__init__.py`?

```
from optuna.study import load_study  # NOQA
```

`optuna.study.create_study` is linked to `optuna.create_study` in [the file](https://github.com/pfnet/optuna/blob/master/optuna/__init__.py#L15), and I would like to keep the consistency of API.

**Comment by sile** [comment]
@toshihikoyanase Thank you for your comment. I imported `load_study` function into top-level module.

**Pull request merged**